src/framework/constraints.ml

```diff
@@ -430,7 +430,13 @@ struct
     lift_fun ctx liftmap S.enter ((|>) args % (|>) f % (|>) r) []
 
   let query ctx (type a) (q: a Queries.t): a Queries.result =
-    lift_fun ctx identity S.query  (fun (x) -> x q)            (Queries.Result.bot q)
+    match q with
+    | Queries.Invariant context ->
+      let e = D.invariant context ctx.local in
+      BatOption.map_default (Queries.ES.singleton) (Queries.ES.top ()) e
```

Comment:
Yes, but is the topmost Lifter that is always there non-conditionally. One could push it down to the `PathSensitive` functors, but there are two different ones and I think they are complicated enough as it stands.
I see two possible alternatives: Rename the `DeadCodeLifter` or introduce a dedicated lifter just for these queries? I'm not sure what would be the better one here?

Reply:
This could even go entirely outside of the functors, right where the `ctx` for transformations is defined in `Control`. Witness generation accesses it directly as well.

Reply:
I now pulled it out. It still has to be a part of the query system though, because I wanted the transfromations to still provide a normal `ask`.